### PR TITLE
Death Flag, Target Setting, Half Damage, Trait Fixes

### PR DESCRIPTION
**Pull request by Vendare**
### Death Flag
By default Foundry sets target flagged with the status effect death to defeated in the combat tracker. In larger fights it is a pain in the ass to flag tokens on the scene as dead by searching through the combat tracker. It is by far faster to set death through the token hud. Death is ommitted from the Conditions in the Effect Tab of the character sheet. An icon with a matching gradiant to the others was added to the Assets
image

###  Target Setting
Loosing all targets after an attack is an annoyance to established GMs and Players alike. Though new Players may find this helpful. To remedy this reseting targets can now be turned off in the System Settings on the Client Side. To aid people new to foundry and AOS:S the Setting is enabled by default

### Half Damage
Under certain conditions damage is halfed before armour is applied Example Attacking a ghost with a none magical weapon. 
An option for this was added to the context menu

### Trait fixes
_Penetrating_: reduced the armour regardless if the target was already at zero armour. This caused penetrating weapons to deal additional damage to none armoured targets

_Ineffective_: Ineffective was too high up the chain in the applyDamage function. Ineffective weapons reduced the targets toughness to zero regardless of inflicted damage and caused a minor wound. The intended effect was to turn all inflicted wounds to minor wounds 

### Language based Skill sorting
As per Request #119 skills are now sorted based by their name on the selected locale.

### CODE REVIEW

File: scripts/actor/actor-aos.js [hunk 1]
```diff
@@ -349,8 +349,9 @@ export class AgeOfSigmarActor extends Actor {
      */
     async applyDamage(damage, {ignoreArmour = false, penetrating = 0, ineffective = false, restraining = false}={}) {
         let armour = this.combat.armour.value
-
-        armour -= penetrating
+        
+        if (armour > 0)
+            armour -= penetrating
```
Comment (moo-man) [inline scripts/actor/actor-aos.js:354]: Wouldn't this still sometimes cause penetrating to deal extra damage? e.g. 1 Armour, with 2 Penetrating = 1 extra damage

File: scripts/system/hooks.js [hunk 2]
```diff
@@ -92,6 +92,15 @@ export default function registerHooks() {
         type: String,
       });
 
+      game.settings.register("age-of-sigmar-soulbound", "looseTarget", {
```
Comment (moo-man) [inline scripts/system/hooks.js:95]: Slight (and common) misspelling, `loose` should be `lose`

File: template/sheet/tab/actor-effects.html [hunk 3]
```diff
@@ -2,17 +2,18 @@
...
-      <a class="condition-toggle">
-        {{#if effect.existing}}
-        <i class="far fa-check-square"></i>
-        {{else}}
-        <i class="far fa-square"></i>
-        {{/if}}
-      </a>
-
-    </div>
+      {{#unless (isNoneSystemCondition effect.key)}}
```
Comment (moo-man) [inline template/sheet/tab/actor-effects.html:5]: I don't think this needs an entire helper, instead use the built in `eq` function.

```html
{{#unless (eq "dead" effect.key)}}
```

File: scripts/actor/actor-aos.js (hunk 1 above)
Comment (Vendare) [reply scripts/actor/actor-aos.js:354]: I wouldn't the penetraiting trait ignores 1 point of armour not more not less. A change is only neccessary if we were to use that value to implement another effect

**Pull request merged**